### PR TITLE
Cleanup the EditorGroundControl component

### PR DESCRIPTION
**Pull request by jsnajdr**
While working on #22617, I did one small fix in the `EditorGroundControl` component (the UI that replaces masterbar when editing a post in "full-screen" mode) and noticed quite a few eyebrow-raising bugs and legacy practices. This PR does a little cleanup of the component.

Each of the 10 commits can be reviewed as a standalone small change.

The main changes are:
- cleanup and bugfix recording analytics events
- stop using `lib/user` in favor of Redux
- stop using global `i18n.translate`

**How to test:**
- check that the post editor works in general
- fake unverified email address by changing the `userNeedsVerification` connected prop and check the behavior for users with unverified emails

### CODE REVIEW

File: client/post-editor/editor-ground-control/index.jsx [hunk 1]
```diff
@@ -216,13 +216,12 @@ export class EditorGroundControl extends PureComponent {
...
 	getCloseButtonPath() {
-		// find the first non-editor path in routeHistory, default to "all posts"
-		const nonEditorPaths = this.props.routeHistory.filter( action => {
-			return ! action.path.match( /^\/(post|page|(edit\/[^\/]+))\/[^\/]+(\/\d+)?$/i );
-		} );
-		return nonEditorPaths && last( nonEditorPaths ) && last( nonEditorPaths ).path
-			? last( nonEditorPaths ).path
-			: this.props.allPostsUrl;
+		// find the last non-editor path in routeHistory, default to "all posts"
+		const lastNonEditorPath = findLast(
```
Comment (samouri) [inline client/post-editor/editor-ground-control/index.jsx]: much better

File: client/post-editor/editor-ground-control/index.jsx [hunk 2]
```diff
@@ -302,18 +302,18 @@ const mapStateToProps = ( state, ownProps ) => {
 };
 
 const mapDispatchToProps = {
-	recordPreviewButtonClick: () =>
-		composeAnalytics(
-			recordTracksEvent(
-				`calypso_editor_${ postUtils.isPage( page ) ? 'page' : 'post' }_preview_button_click`
```
Comment (samouri) [inline client/post-editor/editor-ground-control/index.jsx]: you were right! my eyebrows have been raised

File: client/post-editor/editor-ground-control/index.jsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ import { connect } from 'react-redux';
  */
 import Card from 'components/card';
 import Site from 'blocks/site';
-import * as postUtils from 'lib/posts/utils';
+import { isPage, isPublished } from 'lib/posts/utils';
```
Comment (samouri) [inline client/post-editor/editor-ground-control/index.jsx]: little known fact: webpack is just as good at tree shaking a `*` import as explicit named imports.

File: client/post-editor/editor-ground-control/index.jsx [hunk 4]
```diff
@@ -71,57 +72,6 @@ export class EditorGroundControl extends PureComponent {
...
-				nextProps.userUtils && nextProps.userUtils.needsVerificationForSite( nextProps.site ),
-		} );
-
-		if ( this.props.user ) {
-			this.props.user
-				.off( 'change', this.updateNeedsVerification )
-				.off( 'verify', this.updateNeedsVerification );
-		}
-
-		if ( nextProps.user ) {
```
Comment (samouri) [inline client/post-editor/editor-ground-control/index.jsx]: this `componentWillReceiveProps` was causing a lot of thrashing with subscribing/unsubscribing to `user` on every render!

**Comment by samouri** [comment]
Code changes LGTM. I'll test now

File: client/post-editor/editor-ground-control/index.jsx (hunk 1 above)
Comment (lsinger) [reply client/post-editor/editor-ground-control/index.jsx]: Nice! Thanks!

File: client/post-editor/editor-ground-control/index.jsx (hunk 3 above)
Comment (jsnajdr) [reply client/post-editor/editor-ground-control/index.jsx]: Good to know 👍

**Comment by ockham** [comment]
Drive-by on aba21aa:

> EditorGroundControl doesn't need to be a PureComponent
>
> It's connected to Redux, and `connect` already implements the equivalent
`shouldComponentUpdate` functionality for its wrapped components. Declaring
the component as pure is redundant.

My POV on this is that I still prefer to define the component as a `PureComponent` in order to state the contract that it is fully determined by props (and internal state), even when considered on its own, i.e. without the `connect()` wrapper.

File: client/post-editor/editor-ground-control/index.jsx [hunk 5]
```diff
@@ -294,32 +238,29 @@ export class EditorGroundControl extends PureComponent {
...
-					'Editor',
-					`Clicked Preview ${ postUtils.isPage( page ) ? 'Page' : 'Post' } Button`,
-					`Editor Preview ${ postUtils.isPage( page ) ? 'Page' : 'Post' } Button Clicked`,
-					`editor${ postUtils.isPage( page ) ? 'Page' : 'Post' }ButtonClicked`
-				)
+const mapDispatchToProps = {
+	recordPreviewButtonClick: post => {
+		const postIsPage = isPage( post );
+		return composeAnalytics(
+			recordTracksEvent( `calypso_editor_${ postIsPage ? 'page' : 'post' }_preview_button_click` ),
```
Comment (gwwar) [inline client/post-editor/editor-ground-control/index.jsx]: I understand this was here before, but as a general note, let's avoid   crafting stat strings like this. It makes it so difficult to find actual usages with search later. For example searching for "calypso_editor_post_preview_button_click" would miss the usage in this file.

**Pull request merged**